src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,381 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.decomposition.signed_hahn
+
+/-!
+# Jordan decomposition
+
+This file proves the existence and uniqueness of the Jordan decomposition for signed measures.
+The Jordan decomposition theorem states that, given a signed measure `s`, there exists a
+unique pair of mutually singular measures `μ` and `ν`, such that `s = μ - ν`.
+
+The Jordan decomposition theorem for measures is a corollary of the Hahn decomposition theorem and
+is useful for the Lebesgue decomposition theorem.
+
+## Main definitions
+
+* `measure_theory.jordan_decomposition`: a Jordan decomposition of a measurable space is a
+  pair of mutually singular finite measures. We say `j` is a Jordan decomposition of a signed
+  meausre `s` if `s = j.pos_part - j.neg_part`.
+* `measure_theory.signed_measure.to_jordan_decomposition`: the Jordan decomposition of a
+  signed measure.
+* `measure_theory.signed_measure.to_jordan_decomposition_equiv`: is the `equiv` between
+  `measure_theory.signed_measure` and `measure_theory.jordan_decomposition` formed by
+  `measure_theory.signed_measure.to_jordan_decomposition`.
+
+## Main results
+
+* `measure_theory.signed_measure.to_signed_measure_to_jordan_decomposition` : the Jordan
+  decomposition theorem.
+* `measure_thoery.signed_measure.to_signed_measure_injective` : the Jordan decomposition of a
+  signed measure is unique.
+
+## Tags
+
+Jordan decomposition theorem
+-/
+
+noncomputable theory
+open_locale classical measure_theory ennreal
+
+variables {α β : Type*} [measurable_space α]
+
+namespace measure_theory
+
+/-- A Jordan decomposition of a measurable space is a pair of mutually singular,
+finite measures. -/
+@[ext] structure jordan_decomposition (α : Type*) [measurable_space α] :=
+(pos_part neg_part : measure α)
+[pos_part_finite : finite_measure pos_part]
+[neg_part_finite : finite_measure neg_part]
+(mutually_singular : pos_part ⊥ₘ neg_part)
+
+attribute [instance] jordan_decomposition.pos_part_finite
+attribute [instance] jordan_decomposition.neg_part_finite
+
+namespace jordan_decomposition
+
+open measure vector_measure
+
+variable (j : jordan_decomposition α)
+
+instance : has_zero (jordan_decomposition α) :=
+{ zero := ⟨0, 0, mutually_singular.zero⟩ }
```

Comment:
Can you add the two obvious simp lemmas that say the pos and neg parts of 0 are 0 (and similarly for neg?)